Dockerfile

```diff
@@ -3,9 +3,21 @@ FROM continuumio/miniconda3:23.3.1-0
 
 # Install dependencies
 RUN conda install -c conda-forge google-cloud-sdk && \
-    apt update -y && \
-    apt install rsync vim -y && \
-    rm -rf /var/lib/apt/lists/*
-
-# Install sky
-RUN pip install --no-cache-dir "skypilot[all]==0.6.0"
+    gcloud components install gke-gcloud-auth-plugin --quiet && \
+    find /opt/conda -name 'gke-gcloud-auth-plugin' -type f -exec ln -s {} /usr/local/bin/gke-gcloud-auth-plugin \; && \
+    # Install system packages
+    apt-get update -y && \
+    apt-get install --no-install-recommends -y \
+        git gcc rsync sudo patch openssh-server \
+        pciutils nano fuse socat netcat curl rsync vim && \
+    rm -rf /var/lib/apt/lists/* && \
+    # Install kubectl
+    curl -LO "https://dl.k8s.io/release/v1.28.11/bin/linux/amd64/kubectl" && \
+    install -o root -g root -m 0755 kubectl /usr/local/bin/kubectl && \
+    rm kubectl && \
+    # Install uv and skypilot
+    curl -LsSf https://astral.sh/uv/install.sh | sh && \
+    ~/.local/bin/uv pip install --prerelease=allow skypilot-nightly[all] --system && \
+    # Cleanup all caches to reduce the image size
+    conda clean -afy && \
+    rm -rf ~/.cache/pip ~/.cache/uv
```

Comment:
`uv` has its own clean up command `uv cache clean`. Not sure it will be safer than deleting the directory.

Reply:
Thanks, added that in addition to the removing the cache dirs.